app/assets/javascripts/search-vocab.js

```diff
@@ -0,0 +1,55 @@
+$(document).ready(function() {
+  $('.add-to-vocab-btn').click(function(e) {
+    e.preventDefault();
+    var signId = $(this).attr('data-sign-id');
+
+    var successNotice =
+      '<div class="flash vocab_var_notice ajax-success-response">'
+      + 'You have added a sign to your vocab sheet.'
+      + '</div>';
+
+    var failureNotice =
+      '<div class="flash vocab_var_notice ajax-failure-response">'
+      + 'There was an error adding a sign to your vocab sheet.'
+      + '</div>';
+
+    var vocabList = '.vocab_sheet_bar ul';
+
+    $.ajax({
+      url: '/vocab_sheet/items/',
+      method: 'POST',
+      data: {
+        sign_id: signId,
+      },
+      headers: {
+        'X-CSRF-Token': $('meta[name="authenticity-token"]').attr('content'),
+      },
+    }).done(function(data) {
+     if ($('.vocab_sheet_bar').css('display') === 'none') {
```

Comment:
Do you think this and the failure stuff below could get pulled out into separate functions to tidy up the ajax call?

Reply:
I can do that :)